cmd/wstail/forwarders.go

```diff
@@ -1,106 +1,107 @@
 package main
 
 import (
-	"encoding/json"
 	"fmt"
 	"io/ioutil"
-	"net/http"
-	"os"
-	"os/exec"
 	"path/filepath"
-	"strings"
-
-	"github.com/gravitational/logging-app/lib/forwarders"
 
+	"github.com/ghodss/yaml"
 	"github.com/gravitational/trace"
 	log "github.com/sirupsen/logrus"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
 )
 
-// updateForwarders updates log forwarder configuration and reloads the logging service.
-// It receives new forwarder configuration, updates the configuration and restarts the rsyslog daemon
-// to force it to reload the configuration
-func updateForwarders(w http.ResponseWriter, r *http.Request) (err error) {
-	if r.Method != "PUT" {
-		return trace.BadParameter("invalid HTTP method: %v", r.Method)
+// initLogForwarders reads config map that contains log forwarder resources and creates
+// respective rsyslog configuration files
+func initLogForwarders() error {
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		return trace.Wrap(err)
 	}
-	var forwarders []forwarders.Forwarder
-	if err = readJSON(r, &forwarders); err != nil {
+
+	client, err := kubernetes.NewForConfig(config)
+	if err != nil {
 		return trace.Wrap(err)
 	}
 
-	// Remove previous forwarder configuration files
-	if err = filepath.Walk(rsyslogConfigDir, func(path string, info os.FileInfo, err error) error {
-		if err != nil {
-			return trace.Wrap(err)
-		}
-		if path == rsyslogConfigDir {
-			return nil
-		}
-		if info.IsDir() {
-			return filepath.SkipDir
-		}
-		log.Infof("removing %v", path)
-		return trace.Wrap(os.Remove(path), "failed to remove %v", path)
-	}); err != nil {
-		log.Warningf("failed to delete forwarder configuration files: %v", err)
+	configMap, err := client.ConfigMaps(systemNamespace).Get(
+		forwardersConfigMap, metav1.GetOptions{})
+	if err != nil {
+		return trace.Wrap(err)
```

Comment:
(optional) Could use rigging to map the error

Reply:
Yeah, I was considering it, but decided not to bring in another monstrous dependency just for a single line.